sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/InMemoryCatalog.scala

```diff
@@ -48,7 +48,7 @@ class InMemoryCatalog extends ExternalCatalog {
   private val catalog = new scala.collection.mutable.HashMap[String, DatabaseDesc]
 
   private def filterPattern(names: Seq[String], pattern: String): Seq[String] = {
-    val regex = pattern.replaceAll("\\*", ".*").r
+    val regex = pattern.replaceAll("\\.\\*", "\\*").replaceAll("\\*", "\\.\\*").r
```

Comment:
@andrewor14 If you want to include my changes in your PR, that is fine. I did not see your changes for that though.
@yhuai I think we need to have consistent behavior as Hive does. Hive supports query like "list functions like '\*f\*' " or "list functions like 'f.\*' ", for the first case, our current implementation will cause exception and return empty.